lib/addStyles.js

```diff
@@ -67,7 +70,18 @@ module.exports = function(list, options) {
 	if (!options.singleton) options.singleton = isOldIE();
 
 	// By default, add <style> tags to the <head> element
-	if (!options.insertInto) options.insertInto = "head";
+        if (!options.insertInto) options.insertInto = "head";
+
+        // If provide function in options, then use it for resolve "head" element
+        // Usefull for Shadow Root style i.e
+        // {
+        // insertIntoCallable: "path/to/insertInto.js"
+        //}
+        // path/to/insertInto.js
+        // module.exports = function () { return document.querySelector("#foo").shadowRoot }
+        getElement = options.insertIntoCallable != null
```

Comment:
```js
var element = typeof options.insertInto === 'function' 
  ? getElement(options.insertInto) 
  : getElement(getTarget)
```

Reply:
Another way maybe to pass the `options` to `getTarget` => 

```js
var getTarget = function (target, options) {
    return typeof options.insertInto === 'function' 
      ? options.insertInto() 
      : function (target) { return document.querySelector(target); }
}
```

Something like that